### PR TITLE
Unit tests: Added to cover untested areas of code

### PR DESCRIPTION
**Pull request by neteinstein**

### CODE REVIEW

**Comment by neteinstein** [comment]
@bruno-garces Since you are touching this, travis-ci.org is shuting down and moving to travis-ci.com, I've already taken care of the migration.

Can you please review this PR and approve so we can make the master also work there.

File: rxjava/src/debug/kotlin/com/mindera/skeletoid/rxjava/schedulers/Schedulers.kt [hunk 1]
```diff
@@ -23,4 +23,6 @@ object Schedulers {
     fun io(): Scheduler = io.reactivex.schedulers.Schedulers.from(ioThreadPool)
 
     fun single(): Scheduler = Schedulers.single()
-}
\ No newline at end of file
+
+    fun main(): Scheduler = io.reactivex.android.schedulers.AndroidSchedulers.mainThread()
```
Comment (bruno-garces) [inline rxjava/src/debug/kotlin/com/mindera/skeletoid/rxjava/schedulers/Schedulers.kt:21]:
```
io.reactivex.schedulers.Schedulers
```

Given that we have this import maybe we could either update `computation` and `io` removing the unnecessary full path or update `single` to be coherent with the other and remove the import.

File: rxjava/src/test/java/com/mindera/skeletoid/rxjava/CompletableExtensionsTest.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    // ATTENTION !!!!
+    // This is hitting main in all threadpools since we are defaulting to TRAMPOLINE on QA builds.
+    // It would be great to not do it, but that would break Apps using QA build to UI Tests
+
+    @Test
+    fun testCompletableSubscribeOnIO() {
+        var thread: String? = null
+        Completable.fromAction { thread = Thread.currentThread().name }
+            .subscribeOnIO()
+            .blockingGet()
```
Comment (bruno-garces) [inline rxjava/src/test/java/com/mindera/skeletoid/rxjava/CompletableExtensionsTest.kt]: There is no need to do `blockingGet`

```
Completable.fromAction { thread = currentThreadName() }
    .subscribeOnIO()
    .test()
    .assertNoErrors()
    .assertComplete()
```

File: rxjava/src/test/java/com/mindera/skeletoid/rxjava/CompletableExtensionsTest.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+class CompletableExtensionsTest {
+
+    // ATTENTION !!!!
+    // This is hitting main in all threadpools since we are defaulting to TRAMPOLINE on QA builds.
+    // It would be great to not do it, but that would break Apps using QA build to UI Tests
+
+    @Test
+    fun testCompletableSubscribeOnIO() {
+        var thread: String? = null
+        Completable.fromAction { thread = Thread.currentThread().name }
```
Comment (bruno-garces) [inline rxjava/src/test/java/com/mindera/skeletoid/rxjava/CompletableExtensionsTest.kt]: We could have a small function that would return this

```
private fun currentThreadName(): String = Thread.currentThread().name
```

File: rxjava/src/test/java/com/mindera/skeletoid/rxjava/MaybeExtensionsTest.kt [hunk 4]
```diff
@@ -1,15 +1,85 @@
...
+    // ATTENTION !!!!
+    // This is hitting main in all threadpools since we are defaulting to TRAMPOLINE on QA builds.
+    // It would be great to not do it, but that would break Apps using QA build to UI Tests
+
+    @Test
+    fun testMaybeSubscribeOnIO() {
+        var thread: String? = null
+        Maybe.fromCallable { thread = Thread.currentThread().name }
+            .subscribeOnIO()
+            .blockingGet()
```
Comment (bruno-garces) [inline rxjava/src/test/java/com/mindera/skeletoid/rxjava/MaybeExtensionsTest.kt]: Same here

File: rxjava/src/test/java/com/mindera/skeletoid/rxjava/ObservableExtensionsTest.kt [hunk 5]
```diff
@@ -1,14 +1,92 @@
...
+    fun testObservableSubscribeOnIO() {
+        var thread: String? = null
+        Observable.fromCallable { thread = Thread.currentThread().name }
+            .subscribeOnIO()
+            .single(Unit)
+            .blockingGet()
+
+        println("$thread - ${Thread.currentThread().name}")
+
+        assertEquals(thread, Thread.currentThread().name) // To be fixed: read above
```
Comment (bruno-garces) [inline rxjava/src/test/java/com/mindera/skeletoid/rxjava/ObservableExtensionsTest.kt]: Consider change to the following

```
private fun currentThreadName(): String = Thread.currentThread().name

@Test
fun testObservableSubscribeOnIO() {
    Observable.just(1)
        .map { currentThreadName() }
        .subscribeOnIO()
        .test()
        .assertValueAt(0) { it == "main" }
        .assertValueCount(1)
        .assertNoErrors()
        .assertComplete()
}
```

File: rxjava/src/test/java/com/mindera/skeletoid/rxjava/SingleExtensionsTest.kt [hunk 6]
```diff
@@ -1,12 +1,80 @@
...
+    // It would be great to not do it, but that would break Apps using QA build to UI Tests
+
+    @Test
+    fun testSingleSubscribeOnIO() {
+        var thread: String? = null
+        Single.fromCallable { thread = Thread.currentThread().name }
+            .subscribeOnIO()
+            .blockingGet()
+
+        assertEquals(thread, "main") // To be fixed: read above
```
Comment (bruno-garces) [inline rxjava/src/test/java/com/mindera/skeletoid/rxjava/SingleExtensionsTest.kt]: We could follow the same approach as explained before.

**Review by bruno-garces — CHANGES_REQUESTED** [review]

**Review by bruno-garces — APPROVED** [review]

**Comment by neteinstein** [comment]
continuous-integration/travis-ci Expected — Waiting for status to be reported
Required

This will always be like this as it's the old travis.

Should we merge @bruno-garces ?

**Comment by bruno-garces** [comment]
> continuous-integration/travis-ci Expected — Waiting for status to be reported
> Required
> 
> This will always be like this as it's the old travis.
> 
> Should we merge @bruno-garces ?

Do it

**Pull request merged**